stdlib/ballerina-database/src/main/java/org/ballerinalang/database/sql/SQLDatasourceUtils.java

```diff
@@ -1107,6 +1141,42 @@ public static BStruct createMultiModeDBClient(Context context, String dbType,
                 null);
     }
 
+    private static void registerOutParameter(PreparedStatement stmt, int index, int sqlType, String structuredSQLType,
+            String databaseProductName) throws SQLException {
+        if (databaseProductName.equals(POSTGRES_DATABASE_NAME)) {
+            ((CallableStatement) stmt).registerOutParameter(index + 1, sqlType);
+        } else {
+            ((CallableStatement) stmt).registerOutParameter(index + 1, sqlType, structuredSQLType);
+        }
+    }
+
+    private static void setArrayValue(Object[] arrayValue, Connection conn, PreparedStatement stmt, int index,
+            int sqlType, String databaseProductName, String structuredSQLType) throws SQLException {
+        if (arrayValue == null) {
+            stmt.setNull(index + 1, sqlType);
+        } else {
+            // In POSTGRESQL, need to pass "float8" to indicate DOUBLE value.
+            if (Constants.SQLDataTypes.DOUBLE.equals(structuredSQLType) && POSTGRES_DATABASE_NAME
+                    .equals(databaseProductName)) {
+                structuredSQLType = POSTGRES_DOUBLE;
+            }
+            Array array = conn.createArrayOf(structuredSQLType, arrayValue);
+            stmt.setArray(index + 1, array);
+        }
+    }
+
+    private static void setIntValue(PreparedStatement stmt, int sqlType, int index, Integer val) throws SQLException {
+        if (val == null) {
+            stmt.setNull(index + 1, sqlType);
+        } else if (sqlType == Types.TINYINT) {
+            stmt.setByte(index + 1, val.byteValue());
+        } else if (sqlType == Types.SMALLINT) {
+            stmt.setShort(index + 1, val.shortValue());
+        } else {
+            stmt.setInt(index + 1, val);
+        }
+    }
```

Comment:
We don't need a separate method with different cases here.  Instead of this we need a seperate method called setSmallIntValue and call it from the "SMALLINT" case of setParameter method of the AbstractSQLAction.java class